src/lib/getSelector.ts

```diff
@@ -30,11 +30,8 @@ export const getSelector = (node: Node | null | undefined, maxLen?: number) => {
       const part = el.id
         ? '#' + el.id
         : getName(el) +
-          (el.classList &&
-          el.classList.value &&
-          el.classList.value.trim() &&
-          el.classList.value.trim().length
-            ? '.' + el.classList.value.trim().replace(/\s+/g, '.')
+          (el.classList?.length
```

Comment:
The `.classList` method will always exist in baseline browsers, and its not necessary to check if it's empty because it will just result in an empty array if it is. So you can just do this instead:

```js
getName(el) + Array.from(el.classList).sort().join('.')
```

Reply:
Ah I missed the check for `node` not being `null` (or `undefined`) above and only looked at the top function call (which said it could be one of those).

Fixed now.

Reply:
Don't we still want the `'.'` before the first class name?

Reply:
Ah but only if the `classList` is not empty! That's why we need that check. Reverting the last change.

Reply:
Barry, you're giving up at code golf too easily! You could move the `.join()` call to an outer array to handle the case Rick pointed out:

```js
[getName(el), ...[...el.classList].sort()].join('.')
```

Reply:
<img width="485" alt="Your JS developers were so preoccupied with whether they could do it in one line... they didn't stop to think if they should." src="https://github.com/user-attachments/assets/08008048-12b3-4d15-983f-7c2491acd014">

Reply:
😎

Reply:
the change is replacing the year-old

https://github.com/GoogleChrome/web-vitals/blob/9b932519b16f72328c6d8e9814b811f1bc1a0bb5/src/lib/getSelector.ts#L33-L37

and is still a nested ternary, so let's not pretend it's Hemingway or something :P

Reply:
Oh you were serious about this! Seems more obscure to figure out what's going on but ok.

Reply:
> Seems more obscure to figure out what's going

This is probably subjective based on who's looking at it. Personally I find that calling `.join()` on a list of items is easier to understand than string concatenation.